src/login/logind-dbus.c

```diff
@@ -46,47 +46,77 @@
 #include "utmp-wtmp.h"
 #include "virt.h"
 
-static int get_sender_session(Manager *m, sd_bus_message *message, sd_bus_error *error, Session **ret) {
+static int get_sender_session(
+                Manager *m,
+                sd_bus_message *message,
+                bool try_harder,
```

Comment:
So I thought about this a while, and even had an earlier version of the PR that did something very close to what you suggest, but then I figured that the complexity this brings in is a bit too much, given that this is not as magic as it might appear at first, given that the "display" sesison is pretty well defined.

I'll have another look, maybe we can find another way.